pkg/workloadmeta/collectors/internal/kubeapiserver/node.go

```diff
@@ -8,11 +8,45 @@
 package kubeapiserver
 
 import (
+	"context"
+
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
 
 	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
 )
 
+const nodeStoreName = "node-store"
+
+func init() {
```

Comment:
I think we can avoid using `init` here.

We can have a function in `kubeapiserver.go` that returns all the generators instead. Something like:

```go
func storeGenerators() []storeGenerator{
	var generators []storeGenerator
	
	generators = append(generators, newNodeStore)

	if config.Datadog.GetBool("cluster_agent.collect_kubernetes_tags") {
		generators = append(generators, newPodStore)	
	}
	
	return generators
}
```

Also, moving `if config.Datadog.GetBool("cluster_agent.collect_kubernetes_tags")` out of the generator, we can simplify some code because we can delete the error returned in `type storeGenerator`.